### ISSUE TITLE
Unexplained "error while trying to handle message from plugin"; build and test to "succeed" nonetheless

### ISSUE THREAD

**Issue by dabrahams** [apple/swift-package-manager#7195]
### Description

I get this unexplained error from SPM when building my project, yet `swift test` exits with nonzero exit code.

### Expected behavior

Errors due to plugin execution should stop the build.

### Actual behavior

Build continues; completes with zero exit status.  Error message buried in the output

### Steps to reproduce

1. Check out the spm-unhandled-message tag of https://github.com/hylo-lang/hylo
2. Follow the setup instructions in the README
3. `swift test --explicit-target-dependency-import-check=error`

### Swift Package Manager version/commit hash

Swift Package Manager - Swift 5.9.0

### Swift & OS version (output of `swift --version ; uname -a`)

swift-driver version: 1.87.1 Apple Swift version 5.9 (swiftlang-5.9.0.128.108 clang-1500.0.40.1)
Target: arm64-apple-macosx14.0
Darwin DaveA-MBP16.local 23.1.0 Darwin Kernel Version 23.1.0: Mon Oct  9 21:27:24 PDT 2023; root:xnu-10002.41.9~6/RELEASE_ARM64_T6000 arm64

**Comment by neonichu** [comment]
I agree that these errors seem weird. I *think* the assumption was that if the communication channel is broken at this level, the plugin won't be able to communicate back?

Either way, we should probably rectify that by returning a proper exit code from the plugin so that execution can stop if we encounter these.
